### PR TITLE
meson: allow to customize default mode for tty/pts devices

### PR DESCRIPTION
**Pull request by yuwata**
Closing #35599.

### CODE REVIEW

**Comment by poettering** [comment]
There's also a macro TTY_MODE in src/basic/terminal-util.h which should be made the same too

**Comment by poettering** [comment]
btw, I think we should consider switching the default too. I think "mesg y"/"mesg n" is kinda dump in 2024. Let's switch this over upstream, announce that in NEWS and tell downstreams if they want to return to 1990s unix they should change th ebuild to revert to old bvehaviour

**Comment by yuwata** [comment]
@poettering Thank you for the review. Addressed the requests. PTAL.

File: tools/meson-render-jinja2.py [hunk 1]
```diff
@@ -17,7 +17,9 @@ def parse_config_h(filename):
         if not m:
             continue
         a, b = m.groups()
-        if b and b[0] in '0123456789"':
+        # The function ast.literal_eval() cannot be evaluate octal integers, e.g. 0600.
```
Comment (YHNdnzj) [inline tools/meson-render-jinja2.py]:
```suggestion
        # The function ast.literal_eval() cannot evaluate octal integers, e.g. 0600.
```

Comment (YHNdnzj) [reply tools/meson-render-jinja2.py]: Hmm, we have the following FIXME for dev-kvm-mode:

https://github.com/systemd/systemd/blob/d0a63cf0412ccb9b4edd1f6048e25aa7f68b0133/meson.build#L976

Not sure about the best strategy here.

Comment (yuwata) [reply tools/meson-render-jinja2.py]: That cannot be used at least for TTY_MODE, as it is used as both string and integer.

File: src/basic/terminal-util.h [hunk 2]
```diff
@@ -143,8 +143,8 @@ int vt_release(int fd, bool restore_vt);
 
 void get_log_colors(int priority, const char **on, const char **off, const char **highlight);
 
-/* This assumes there is a 'tty' group */
-#define TTY_MODE 0620
+/* Assume TTY_MODE is defined in config.h. Also, this assumes there is a 'tty' group. */
+assert_cc((TTY_MODE & ~0666) == 0);
```
Comment (poettering) [inline src/basic/terminal-util.h:147]: maybe also do: 

```
assert_cc((TTY_MODE & 0711) == 0600);
```

Comment (yuwata) [reply src/basic/terminal-util.h:147]: That should be redundant as it is already checked that the string starts '06' in meson. But OK. I will add that.

**Pull request merged**